pkg/comp-functions/functions/vshnpostgres/user_management.go

```diff
@@ -38,17 +40,28 @@ func UserManagement(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, svc *runti
 
 	for _, access := range comp.Spec.Parameters.Service.Access {
 
-		userPasswordRef := addUser(comp, svc, *access.User)
-
 		dbuser := *access.User
 		dbname := dbuser
 		if access.Database != nil {
 			dbname = *access.Database
 		}
 
-		addDatabase(comp, svc, dbuser, dbname)
+		userPasswordRef := addUser(comp, svc, *access.User)
+		roleResourceName := fmt.Sprintf("%s-%s-role", comp.GetName(), dbuser)
+
+		dbResourceName := fmt.Sprintf("%s-%s-database", comp.GetName(), dbname)
+		if isResourceSyncedAndReady(svc, roleResourceName) {
+			addDatabase(comp, svc, dbuser, dbname)
+		} else {
+			svc.SetDesiredResourceReadiness(dbResourceName, runtime.ResourceUnReady)
+		}
 
-		addGrants(comp, svc, dbuser, dbname, access.Privileges)
+		grantResourceName := fmt.Sprintf("%s-%s-%s-grants", comp.GetName(), dbuser, dbname)
+		if isResourceSyncedAndReady(svc, dbResourceName) {
+			addGrants(comp, svc, dbuser, dbname, access.Privileges)
+		} else {
+			svc.SetDesiredResourceReadiness(grantResourceName, runtime.ResourceUnReady)
```

Comment:
There might be an edgecase where this doesn't work. If the resource doesn't exist at all currently it will not be able to set it to unready.

It makes more sense to set the composite to unready.